pkg/cmd/run/shared/shared.go

```diff
@@ -424,19 +425,31 @@ func preloadWorkflowNames(client *api.Client, repo ghrepo.Interface, runs []Run)
 }
 
 type JobsPayload struct {
-	Jobs []Job
+	TotalCount int `json:"total_count"`
+	Jobs       []Job
 }
 
 func GetJobs(client *api.Client, repo ghrepo.Interface, run *Run) ([]Job, error) {
 	if run.Jobs != nil {
 		return run.Jobs, nil
 	}
-	var result JobsPayload
-	if err := client.REST(repo.RepoHost(), "GET", run.JobsURL, nil, &result); err != nil {
-		return nil, err
+
+	v := url.Values{}
+	v.Set("per_page", "100")
+
+	jobsPath := fmt.Sprintf("%s?%s", run.JobsURL, v.Encode())
+
+	for jobsPath != "" {
+		var resp JobsPayload
+		var err error
+		jobsPath, err = client.RESTWithNext(repo.RepoHost(), http.MethodGet, jobsPath, nil, &resp)
+		if err != nil {
+			return run.Jobs, err
```

Comment:
Similar `cli/cli` code returns nothing if any error is raised like the original code.

```suggestion
			return nil, err
```

Reply:
@andyfeller Got it. Along those same lines, does it make sense to reset `run.Jobs` back to `nil` if there is an error?
https://github.com/harveysanders/cli/blob/e96fe1d007f401db6660dcf250d33c69a7c8ae8d/pkg/cmd/run/shared/shared.go#L433-L435

Or would you prefer to use an internal `Job`s slice and only update `run.Jobs` on successful completion?
```go
var results []Job
// fetch jobs
// ....
results = append(results, resp.Jobs...)
// ....
// all fetching complete
run.Jobs = results
return run.Jobs, err
```

Reply:
I opted for the former
https://github.com/cli/cli/pull/7879/commits/6f0f5a89e2538c20cb02b3e1951026f08ea1e082